Link to YARP::YARP_eigen

Compiling the baseEstimatorV1 device against a YARP used from the build directory result in the following error:
~~~
[ 45%] Building CXX object devices/baseEstimatorV1/CMakeFiles/baseEstimatorV1.dir/src/baseEstimatorV1.cpp.o
In file included from /usr/local/src/robot/robotology-superbuild/robotology/whole-body-estimators/devices/baseEstimatorV1/src/baseEstimatorV1.cpp:9:0:
/usr/local/src/robot/robotology-superbuild/robotology/whole-body-estimators/devices/baseEstimatorV1/include/baseEstimatorV1.h:32:10: fatal error: yarp/eigen/Eigen.h: No such file or directory
 #include <yarp/eigen/Eigen.h>
~~~
this happens because `baseEstimatorV1` does not correctly links to `YARP::YARP_eigen`.

--- a/devices/baseEstimatorV1/CMakeLists.txt
+++ b/devices/baseEstimatorV1/CMakeLists.txt
@@ -8,6 +8,11 @@ find_package(iDynTree ${iDynTree_VERSION} REQUIRED)
 
 find_package(ICUB REQUIRED)
 
+find_package(YARP REQUIRED COMPONENTS
+             eigen
+             os
+             sig)
+
 set(FBE_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/baseEstimatorV1.h
                 ${CMAKE_CURRENT_SOURCE_DIR}/include/WalkingLogger.tpp
                 ${CMAKE_CURRENT_SOURCE_DIR}/include/WalkingLogger.hpp
@@ -37,6 +42,9 @@ target_include_directories(baseEstimatorV1 PRIVATE
                            ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 target_link_libraries(baseEstimatorV1 PUBLIC ${YARP_LIBRARIES}
+                                      YARP::YARP_eigen
+                                      YARP::YARP_os
+                                      YARP::YARP_sig
                                       ${iDynTree_LIBRARIES}
                                       floatingBaseEstimationRPC-service
                                       PRIVATE Eigen3::Eigen)

